third_party/blink/renderer/core/dom/scripted_idle_task_controller.cc

```diff
@@ -37,17 +37,16 @@ class IdleRequestCallbackWrapper
       scoped_refptr<IdleRequestCallbackWrapper> callback_wrapper,
       base::TimeTicks deadline) {
 
-    if (!recordreplay::AreEventsDisallowed())
```

Comment:
Can we add back this test?  Regardless of how the controller treats warnings generated from assertions while events are disallowed we still should avoid generating those warnings when we've reviewed them and determined they won't cause later problems.

Reply:
@bhackett1024 
I did not find any evidence that this is ever called non-deterministically.

We never see any warnings of this kind, even though I added an unprotected `Assert` to `Cancel`, which is called directly from here ([src](https://github.com/replayio/chromium/blob/74a7fc46504efc827dbbbdac8daf6c2c772d822f/third_party/blink/renderer/core/dom/scripted_idle_task_controller.cc#L63)).
We see it pop up [here](http://admin/crash/controller/e78f94fa-85c7-4d49-8e93-a65358915d8c), and no warnings. Have not seen any other reports where we were warned about this either.

Can we leave the `if`s out?

Reply:
OK, that's fine.  It looks like these were originally added with the "if" statement with no supporting evidence in 1d719e0466dd326034a79e35f8d378e0248f25d4 ?  It would be good to only have the "if" if we've actually seen or believe there will be problems there, otherwise they will add confusion.